app/src/main/java/com/example/newsapp/domain/PasscodeUseCases.kt

```diff
@@ -32,4 +32,22 @@ class SkipPasscodeUseCase @Inject constructor(
     suspend operator fun invoke(passcode: String) = withContext(ioDispatcher) {
         userRepository.saveUser(User(isLocked = false, passcode = ""))
     }
+}
+
+class IsPasscodeSkipUseCase @Inject constructor(
+    private val userRepository: UserRepository,
+    private val ioDispatcher: CoroutineDispatcher = Dispatchers.IO
+) {
+    suspend operator fun invoke(passcode: String): Boolean = withContext(ioDispatcher) {
+        !userRepository.getUserInfo()!!.isLocked
```

Comment:
Avoid using the !! operator. This is a hack in most cases used for components with a lifecycle.
Instead check is it's equal to false:
`userRepository.getUserInfo()?.isLocked == false`